ray_lightning/ray_ddp.py

```diff
@@ -116,10 +121,19 @@ def __init__(self,
             parallel_devices=[],
             cluster_environment=RayEnvironment(world_size=num_workers),
             **ddp_kwargs)
+        resources_per_worker = resources_per_worker if resources_per_worker \
+            else {}
         self.nickname = "ddp_ray"
         self.num_workers = num_workers
-        self.num_cpus_per_worker = num_cpus_per_worker
-        self.use_gpu = use_gpu
+        self.num_cpus_per_worker = resources_per_worker.pop(
+            "CPU", num_cpus_per_worker)
+
+        if "GPU" in resources_per_worker:
+            self.num_gpus_per_worker = resources_per_worker.pop("GPU")
```

Comment:
Can probably do some validation here (e.g. fractional values) but might not be necessary.

Reply:
This should still work even if it's fractional right?

Reply:
Ah that's right, I guess you can.